ibridges/irodsconnector/tickets.py

```diff
@@ -43,13 +46,72 @@ def create_ticket(self, obj_path: str, expiry_string: str = '') -> tuple:
             (str, bool)
 
         """
-        ticket = irods.ticket.Ticket(self.sess_man.irods_session)
-        ticket.issue('read', obj_path)
-        logging.info('CREATE TICKET: %s: %s', ticket.ticket, obj_path)
+        ticket = irods.ticket.Ticket(self.session.irods_session)
+        ticket.issue(ticket_type, obj_path)
         expiration_set = False
-        if expiry_string != '':
+        if expiry_string is not None:
             try:
                 expiration_set = ticket.modify('expire', expiry_string) == ticket
             except Exception as error:
-                logging.error('Could not set expiration date: %r', error)
+                self.delete_ticket(ticket)
+                raise Exception('Could not set expiration date: %r', error)
         return ticket.ticket, expiration_set
+
+    @property
+    def all_ticket_strings(self) -> list:
+        return [name for name, _, _, _ in self.all_tickets()]
+
+    def get_ticket(self, ticket_str: str) -> irods.ticket.Ticket:
+        if ticket_str in self.all_ticket_strings:
+            return irods.ticket.Ticket(self.session.irods_session, ticket=ticket_str)
+        else:
+            warnings.warn("Ticket does not exist.")
+        return None
+
+    def delete_ticket(self, ticket: irods.ticket.Ticket):
+        if ticket and ticket.string in self.all_ticket_strings:
+            ticket.delete()
+            self.all_tickets(update=True)
+        else:
+            warnings.warn("Ticket does not exist.")
+
+    def all_tickets(self, update: bool = False) -> list:
```

Comment:
You can hint more precisely what the list contains, I guess in this case it is `list[tuple[str,str?,...,]]`